src/dhtrunner.cpp

```diff
@@ -154,21 +155,28 @@ DhtRunner::run(const SockAddr& local4, const SockAddr& local6, const DhtRunner::
             std::cerr << "Can't start peer discovery (IPv6): " << e.what() << std::endl;
         }
     }
+
+    current_node_netid_ = config.dht_config.node_config.network;
+    NodeInsertionPack adc;
+    adc.nid_ = current_node_netid_;
+    adc.node_port_ = getBoundPort();
+    adc.nodeid_ = dht_->getNodeId();
+    msgpack::sbuffer sbuf_node_v4;
+    msgpack::sbuffer sbuf_node_v6;
+    msgpack::pack(sbuf_node_v4, adc);
+    msgpack::pack(sbuf_node_v6, adc);
+
     if (config.peer_discovery) {
-        using sig = void (DhtRunner::*)(const InfoHash&, const SockAddr&);
         if (peerDiscovery4_)
-            peerDiscovery4_->startDiscovery(std::bind(static_cast<sig>(&DhtRunner::bootstrap), this,
-                                                   std::placeholders::_1,std::placeholders::_2));
-
+            peerDiscovery4_->startDiscovery(pack_type_,callbackmap_[pack_type_]);
         if (peerDiscovery6_)
-            peerDiscovery6_->startDiscovery(std::bind(static_cast<sig>(&DhtRunner::bootstrap), this,
-                                                    std::placeholders::_1,std::placeholders::_2));
+            peerDiscovery6_->startDiscovery(pack_type_,callbackmap_[pack_type_]);
     }
     if (config.peer_publish) {
         if (peerDiscovery4_)
-            peerDiscovery4_->startPublish(dht_->getNodeId(), getBoundPort(AF_INET));
+            peerDiscovery4_->startPublish(pack_type_, std::move(sbuf_node_v4));
```

Comment:
avoid moving and simply copy the buffer (use a single buffer instead of sbuf_node_v4, sbuf_node_v6)

Reply:
Hi, adrien,  forget to mention that the msgpack does not allow the copy, and I need the source to be stored in std::map<std::string, msgpack::sbuffer> messages to send, so these two resources may be required?